src/testdir/test_edit.vim

```diff
@@ -0,0 +1,866 @@
+  set noinsertmode
+  " 3) CTRL-\ CTRL-O
+  call setline(1, ['ABC', 'ZZZ', 'def', 'ghi'])
+  call cursor(1, 1)
+  call feedkeys("A\<c-o>ix", 'txin')
+  call assert_equal(['ABxC', 'ZZZ', 'def', 'ghi'], getline(1,'$'))
+  call feedkeys("A\<c-\>\<c-o>ix", 'txin')
+  call assert_equal(['ABxCx', 'ZZZ', 'def', 'ghi'], getline(1,'$'))
+  " 4) CTRL-\ a (should be inserted literally, not special after <c-\>
+  call setline(1, ['ABC', 'ZZZ', 'def', 'ghi'])
+  call cursor(1, 1)
+  call feedkeys("A\<c-\>a", 'txin')
+  call assert_equal(["ABC\<c-\>a", 'ZZZ', 'def', 'ghi'], getline(1, '$'))
+  bw!
+endfunc
+
+func! Test_edit_10()
+  " Test for starting selectmode
+  new
+  set selectmode=key keymodel=startsel
+  call setline(1, ['abc', 'def', 'ghi'])
+  call cursor(1, 4)
+  call feedkeys("A\<s-home>start\<esc>", 'txin')
+  call assert_equal(['startdef', 'ghi'], getline(1, '$'))
+  set selectmode= keymodel=
+  bw!
+endfunc
+
+func! Test_edit_11()
+  " Test that indenting kicks in
+  new
+  set cindent
+  call setline(1, ['{', '', ''])
+  call cursor(2, 1)
+  call feedkeys("i\<c-f>int c;\<esc>", 'tnix')
+  call cursor(3, 1)
+  call feedkeys("i/* comment */", 'tnix')
+  call assert_equal(['{', "\<tab>int c;", "/* comment */"], getline(1, '$'))
+  " added changed cindentkeys slightly
+  set cindent cinkeys+=*/
+  call setline(1, ['{', '', ''])
+  call cursor(2, 1)
+  call feedkeys("i\<c-f>int c;\<esc>", 'tnix')
+  call cursor(3, 1)
+  call feedkeys("i/* comment */", 'tnix')
+  call assert_equal(['{', "\<tab>int c;", "\<tab>/* comment */"], getline(1, '$'))
+  " Use indentexpr instead of cindenting
+  func! Do_Indent()
+    if v:lnum == 3
+      return 3*shiftwidth()
+    else
+      return 2*shiftwidth()
+    endif
+  endfunc
+  setl indentexpr=Do_Indent() indentkeys+=*/
+  call setline(1, ['{', '', ''])
+  call cursor(2, 1)
+  call feedkeys("i\<c-f>int c;\<esc>", 'tnix')
+  call cursor(3, 1)
+  call feedkeys("i/* comment */", 'tnix')
+  call assert_equal(['{', "\<tab>\<tab>int c;", "\<tab>\<tab>\<tab>/* comment */"], getline(1, '$'))
+  set cinkeys&vim indentkeys&vim
+  set nocindent indentexpr=
+  delfu Do_Indent
+  bw!
+endfunc
+
+func! Test_edit_CR()
+  " Test for <CR> in insert mode
+  " basically only in quickfix mode ist tested, the rest
+  " has been taken care of by other tests
+  if !has("quickfix")
+    return
+  endif
+  botright new
+  call writefile(range(1, 10), 'Xqflist.txt')
+  call setqflist([{'filename': 'Xqflist.txt', 'lnum': 2}])
+  copen
+  set modifiable
+  call feedkeys("A\<cr>", 'tnix')
+  call assert_equal('Xqflist.txt', bufname(''))
+  call assert_equal(2, line('.'))
+  cclose
+  botright new
+  call setloclist(0, [{'filename': 'Xqflist.txt', 'lnum': 10}])
+  lopen
+  set modifiable
+  call feedkeys("A\<cr>", 'tnix')
+  call assert_equal('Xqflist.txt', bufname(''))
+  call assert_equal(10, line('.'))
+  bw!
+  lclose
+  call delete('Xqflist.txt')
+endfunc
+
+func! Test_edit_CTRL_()
+  " disabled for Windows builds, why?
```

Comment:
I don't know. I tried setting encoding explicitly to `utf-8` but that does not work. I am possibly missing some ifdef feature or something. See [this](https://ci.appveyor.com/project/chrisbra/vim/build/2748/job/ij13l2kq6lq3cs66#L2253) and [that](https://ci.appveyor.com/project/chrisbra/vim/build/2756/job/1aw4fi06jw46d76y#L1365). The second one is especially mysterious, since at the specified line encoding is reset and all `:bw` commands use the `!` attribute. So for now I have disabled this test, until the test_edit script is finished and then will check again.